cerealize/cerealize.h

```diff
@@ -140,84 +140,124 @@ WithBaseType(const ENTRY& object) {
 // Enumeration for compile-time format selection.
 enum class CerealFormat { Default = 0, Binary = 0, JSON };
 
-// Templated stream types.
-template <CerealFormat>
-struct CerealStreamType {};
-
-template <>
-struct CerealStreamType<CerealFormat::Binary> {
-  typedef cereal::BinaryInputArchive Input;
-  typedef cereal::BinaryOutputArchive Output;
-  template <typename OSTREAM>
-  inline static Output CreateOutputArchive(OSTREAM& os) {
-    return Output(os);
+class CerealFileAppenderBase {
+ public:
+  explicit CerealFileAppenderBase(const std::string& filename, bool append = true)
+       : fo_(filename, (append ? std::ofstream::app : std::ofstream::trunc) | std::ofstream::binary) {
+    long p = fo_.tellp();
+    if (p >= 0) {
+      initial_stream_position_ = static_cast<size_t>(p);
+    } else {
+      BRICKS_THROW(bricks::CerealizeFileStreamErrorException());
+    }
   }
-};
 
-template <>
-struct CerealStreamType<CerealFormat::JSON> {
-  typedef cereal::JSONInputArchive Input;
-  typedef cereal::JSONOutputArchive Output;
-  template <typename OSTREAM>
-  inline static Output CreateOutputArchive(OSTREAM& os) {
-    // TODO(dkorolev): Add NoIndent as a policy once that cereal change is merged.
-    return Output(os, cereal::JSONOutputArchive::Options::NoIndent());
+  inline size_t EntriesWritten() { return entries_written_; }
+  inline size_t BytesWritten() {
```

Comment:
Please make this and the previous method `const`. It might require `std::ostream fo_` -> `mutable std::ofstream fo_`.